### PR TITLE
BaseRepository Created. Category Repository extends base repository

### PR DESCRIPTION
**Pull request by Maruf520**

### CODE REVIEW

File: src/E-Shop.Api/Startup.cs [hunk 1]
```diff
@@ -43,6 +44,7 @@ public void ConfigureServices(IServiceCollection services)
                  optionsBuilder.MigrationsAssembly("E-Shop.Api")));
             services.AddScoped<ICategoryRepository,CategoryRepository>();
             services.AddScoped<ICategoryServices,CategoryServices>();
+            services.AddScoped(typeof(IBaseRepository<>),typeof(BaseRepository<>));
```
Comment (rafiulgits) [inline src/E-Shop.Api/Startup.cs:47]: why using `typeof`? we can use `services.AddScoped(IBaseRepository,BaseRepository);`

File: src/E-Shop.Repositories/BaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using E_Shop.Models;
+using E_Shop.Models.Shared;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace E_Shop.Repositories
```
Comment (rafiulgits) [inline src/E-Shop.Repositories/BaseRepository.cs:10]: rename the namespace root name to `EShop`

**Review by rafiulgits — COMMENTED** [review]
have a look on my comments

File: src/E-Shop.Api/Startup.cs (hunk 1 above)
Comment (Maruf520) [reply src/E-Shop.Api/Startup.cs:47]: I'm getting error when i set services.AddScoped(IBaseRepository,BaseRepository);.
using typeof solves my issue.Thats why i used typeof

File: src/E-Shop.Repositories/BaseRepository.cs (hunk 2 above)
Comment (Maruf520) [reply src/E-Shop.Repositories/BaseRepository.cs:10]: Ok

File: src/E-Shop.Api/Startup.cs (hunk 1 above)
Comment (rafiulgits) [reply src/E-Shop.Api/Startup.cs:47]: ok

**Pull request merged**